msg/tools/generate_microRTPS_bridge.py

```diff
@@ -337,77 +429,279 @@ def generate_agent(out_dir):
         for msg_file in classifier.msgs_to_send:
             if gen_idl:
                 if out_dir != agent_out_dir:
-                    px_generate_uorb_topic_files.generate_idl_file(msg_file, msg_dir, "", os.path.join(out_dir, "/idl"), urtps_templates_dir,
-                                                                   package, px_generate_uorb_topic_files.INCL_DEFAULT, fastrtpsgen_version, ros2_distro, classifier.msg_id_map)
+                    px_generate_uorb_topic_files.generate_idl_file(
```

Comment:
All these if cases are hard to read. Some smaller functions would be nice.

Reply:
yeah I agree. I will think of a way of doing that and issue a PR for it.